iothub/device/src/IotHubBaseClient.cs

```diff
@@ -558,14 +558,14 @@ private protected static ClientPipelineBuilder BuildPipeline()
             return !isFound ? default : (T)handler;
         }
 
-        internal async Task OnMessageReceivedAsync(Message message)
+        internal async Task<MessageAcknowledgement> OnMessageReceivedAsync(Message message)
         {
             if (Logging.IsEnabled)
                 Logging.Enter(this, message, nameof(OnMessageReceivedAsync));
 
             if (message == null)
             {
-                return;
+                return MessageAcknowledgement.Abandon;
```

Comment:
Will this work for MQTT?

Reply:
Good point. For MQTT, we may as well complete here instead